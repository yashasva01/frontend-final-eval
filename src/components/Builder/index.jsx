import * as React from 'react';
import CreateNewType from '../CreateNewType';
import FieldAdder from '../FieldAdder';

function Builder({collectionTypes, setCollectionTypes}) {
  const [currentType, setCurrentType] = React.useState(collectionTypes[0]);
  React.useEffect(() => {
  }, [collectionTypes]);
  return (
    <div className="content"> 
      <div className="contentHeader"> <h1> Content Types </h1> </div>
      <div className="ContentMaker">  
        <CreateNewType collectionTypes={collectionTypes} setCollectionTypes={setCollectionTypes} 
          currentType={currentType} setCurrentType={setCurrentType}/>
        <FieldAdder currentType={currentType} setCurrentType={setCurrentType}/>
      </div> 
    </div>
  );
}

export default Builder;