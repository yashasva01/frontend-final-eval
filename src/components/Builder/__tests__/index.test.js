import Builder from '../index';
import * as React from 'react';
import {render} from '@testing-library/react';

const mockCollection = ['Location', 'Hotel', 'Flight', 'Car'];
const mockSetCollection = jest.fn();

describe('Builder', () => {
  it('should render correctly', () => {
    const {container} = render(<Builder collectionTypes={mockCollection} setCollectionTypes={mockSetCollection}/>);
    expect(container).toMatchSnapshot();
  });
});