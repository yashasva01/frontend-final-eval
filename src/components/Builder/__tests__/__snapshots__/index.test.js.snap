// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Builder should render correctly 1`] = `
<div>
  <div
    class="content"
  >
    <div
      class="contentHeader"
    >
       
      <h1>
         Content Types 
      </h1>
       
    </div>
    <div
      class="ContentMaker"
    >
      <div
        class="contentType"
      >
        <div
          class="searchType"
        >
          <p>
            4 Types
          </p>
          <img
            alt="searchIcon"
            src="icon-search-dark.png"
          />
        </div>
        <div
          class="addNewType"
        >
          <p>
             + NEW TYPE 
          </p>
        </div>
        <div
          class="type"
        >
          <p
            class="contentTypeName"
          >
            Location
          </p>
          <p
            class="contentInstancesName"
          >
             
            0
             
          </p>
        </div>
        <div
          class="type"
        >
          <p
            class="contentTypeName"
          >
            Hotel
          </p>
          <p
            class="contentInstancesName"
          >
             
            0
             
          </p>
        </div>
        <div
          class="type"
        >
          <p
            class="contentTypeName"
          >
            Flight
          </p>
          <p
            class="contentInstancesName"
          >
             
            0
             
          </p>
        </div>
        <div
          class="type"
        >
          <p
            class="contentTypeName"
          >
            Car
          </p>
          <p
            class="contentInstancesName"
          >
             
            0
             
          </p>
        </div>
      </div>
      <div
        class="contentField"
      >
        <div
          class="contentField__header"
        >
          <h2
            class="contentField__title"
          >
             
            Location
             
          </h2>
          <img
            alt="edit logo"
            src="user-edit-text-message-note.png"
          />
        </div>
        <p>
           
          0
           Fields 
        </p>
        <div
          class="fieldAdder"
        >
          <div
            class="addNewField"
          >
            <p>
               Add Another Field 
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
